accepted/single-file/design.md

```diff
@@ -101,22 +127,26 @@ On Startup, the [host components](https://github.com/dotnet/core-setup/blob/mast
   
   * Implements a bundle-probe function, which will be used by the runtime to probe contents of the bundle when resolving assemblies. 
   
-    * Bundle probe has the following signature:
+    ```C++
+  /// <summary>
+    /// <param name="path"> Relative-path to the file being probed. </param>
+    /// <param name="size"> Out-param: size of the file, if found. </param>
+    /// <param name="path"> Out-param: offset within the bundle, if found</param>
+    /// <returns> true if the requested file is found in the bundle, 
+    ///           false otherwise. </returns>
+    /// </summary>
+    bool bundle_probe(const char *path, int64_t *size, int64_t *offset);
+    ```
+    
+  * The `bundle_probe` function pointer is passed to the runtime (encoded as a string) through a property named `BUNDLE_PROBE`.
+  
+  * When probing for assemblies, the the [assembly resolution logic](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/docs/design/features/assembly-conflict-resolution.md#probe-ordering) will treat bundled assemblies similar to assemblies in the app directory. The probe ordering will be in the order:  servicing location, shared store, framework directory(s) from higher to lower, the *single-file bundle*, app directory, additional locations specified in runtimeconfig.dev.json file.
   
-      ```C++
-      /// <summary>
-      /// <param name="path"> Relative-path to the file being probed. </param>
-      /// <param name="size"> Out-param: size of the file, if found. </param>
-      /// <param name="path"> Out-param: offset within the bundle, if found</param>
-      /// <returns> true if the requested file is found in the bundle, 
-      ///           false otherwise. </returns>
-      /// </summary>
-      bool bundle_probe(const char *path, int64_t *size, int64_t *offset);
-      ```
+  * The assemblies on disk are listed in `TRUSTED_PLATFORM_ASSEMBLIES` using absolute paths. The assemblies to be loaded from the single-file bundle will be listed in `TRUSTED_PLATFORM_ASSEMBLIES` using relative paths (as a distinguishing convention). 
```

Comment:
Please make it super clear to what the relative paths are "relative". I know it's a bit weird, but it's important. Not sure if we describe the fact that the bundle stores a directory structure and the paths are effectively relative to the bundle root. But we probably should. And maybe even mention how does that map to the app's layout when built without single-file.

Reply:
OK I'll add a note about recording relative-paths of file names in the meta-data of the bundle.

Actually, since the contents of the bundle always mask the contents of the app-directory, we need not actually store relative paths in the bundle. We could equivalently store absolute paths relative to the app-directory (which will be `AppContext.BaseDirectory`.  In the single-file prototype, this was a convenience, because many parts of the runtime don;t like to handle relative paths. 

However, writing relative paths in the TPA list makes the contents coming from the bundle evident for visual inspection.